### ISSUE TITLE
Update Cluster tools (k-truss, hierarchical) to run in Interactive Enabled mode to make it obvious "Cluster" is working

### ISSUE THREAD

**Issue by GammaVel** [constellation-app/constellation#1089]
### Prerequisites

- [X] Put an X between the brackets on this line if you have done all of
    the following:

    -   Running the latest version of Constellation

    -   Attached the ***Support Package*** via `Help` &gt;
        `Support Package`

    -   Checked the FAQs:
        https://github.com/constellation-app/constellation/wiki/FAQ

    -   Checked that your issue isn’t already filed:
        https://github.com/constellation-app/constellation/issues

    -   Checked that there is not already a module that provides the
        described functionality:
        https://github.com/constellation-app/constellation/wiki/Catalogue-of-Repositories

### Description

For the cluster tools (k-truss, hierarchical) - hitting "cluster" does it's job, but starts in the "non-interactive" mode, which makes it look like nothing happened to the end user (as the display on the screen doesn't change)

### Steps to Reproduce

1.  Open up a graph that has triangular relationships. I have used domestic_airlines_on_time_performance.star which is available from Training Exercise 5 on Github 
2.  Go to Tools Cluster > K Truss or Hierarchical option
3. Click Cluster Button

**Expected behaviour:** Cluster completes and **Interactive - Enabled Button** becomes enabled so the user can see the results

**Actual behaviour:** The cluster completes however the Interactive -Disabled button is enabled so it looks like the clustering has not performed any actions. As a new user this does not show case the functionality so I don't see the changes in the graph. 
![image](https://user-images.githubusercontent.com/83679004/120743279-1078e480-c53c-11eb-82ce-bdd8bfe06a4b.png)

**Reproduces how often:** 100%

### Additional Information
NA

**Comment by GammaVel** [comment]
![image](https://user-images.githubusercontent.com/83679004/120743513-841af180-c53c-11eb-88e1-9d27e96ee86f.png)
This is what it looks like after clicking on Interactive Disabled

**Comment by Atlas139mkm** [comment]
After investigating for a few hours, this doesn't appear to be as easy of a fix as I would have thought. The way that this class is designed doesn't allow me to simply activate/enable the interactive elements when a user clicks the "cluster" button.

![image](https://user-images.githubusercontent.com/88649439/133959142-4289ee93-c108-4331-be56-cd1cc1e9f286.png)

The first action is the action that occurs on the "cluster" button, the second is on the "Interactive - enable/disable" button.

To solve this problem my solution was to add the "Interactive - enable/disable" buttons functionality to the "cluster" button. This way the elements would immediately be enabled when the user presses the "cluster" button. However my current problem is that  the state is null and doesn't get set until a later point so placing the updateInteractivity(); function within the "cluster" button throws an error. I can not figure out a way to 100% ensure all the necessary set up is done before calling updateInteractivity();

What I find strange I can press the "cluster" button and the "Interactive - enable/disable" button directly after and have no issues. However placing the functionality of the "Interactive - enable/disable" button within the "cluster" buttons after the existing code throws a null pointer error. This confuses me as I believe there should be no other processes running in-between the 2 button clicks which would cause the end result to be any different however it is.. @arcturus2 any thoughts or ideas?

I tried the below however it did not work either.
Future<?> F = PluginExecution.withPlugin(new KTrussCalculatePlugin(interactiveButton.isSelected())).executeLater(graph);
        try {
            F.get();

**Comment by Atlas139mkm** [comment]
Adding on to my previous comment, If this is too difficult to automate the activating of the interactivity, my other thought would be to increase the visibility of the the "Interactive - enable/disable" button. This could be done to draw the users attention to it, to indicate that clicking that button is the next step in the interaction. This could be done but changing the colour of the button or increasing the font weight and size?

**Comment by arcturus2** [comment]
I will try to respond to your first question too but I think we can make it easier to know that "Interactive" button is what needs to be clicked. How about bolding the text of that button to start with?

Going off on a tangent but I want to look at the idea of having "primary" and "secondary" buttons and they would be styled as such. For example even on Github the most common next step is coloured and the rest are gray.

![image](https://user-images.githubusercontent.com/39325530/134114942-82cc2e35-4d05-4a04-bf16-00694a2ff17e.png)

This idea is something I am taking from working with Twitter Bootstrap (https://getbootstrap.com/docs/5.1/components/buttons/). I'd like to be able to assign a button as primary and have it coloured. If we do this then we probably need to do it holistically across the application otherwise it will look odd. Probably another ticket though. Thoughts?

**Comment by Atlas139mkm** [comment]
@arcturus2 YES! completely agree. I think the use of colour is great for constellation. I have found a lot of the application sticks with either plain black or white buttons. It definetly guides the user a lot more, drawing their attention to the constructive elements / next logical steps.  I think my original plan for the notes view redo was to colour buttons based on their functionality. eg red for delete note, green for add nodes, yellow for remove nodes. traffic light style system. Those bootstrap buttons are a perfect example.

**Comment by Atlas139mkm** [comment]
@arcturus2 Purely just a proof of idea. The colours and the contrast are not good for the button here. But it just shows how it helps the user more so than it did before.

![ktrussInteractionButton](https://user-images.githubusercontent.com/88649439/134457436-9880a50d-7135-431a-bbd4-fc7deb7e5a1f.gif)

**Comment by arcturus2** [comment]
Hmm understood. I think the text colour needs to be white and perhaps bold.
What does it look like without adding colour but just bolding the text only?
I think having the text saying "enabled" and "disabled" is a good addition.

**Comment by Atlas139mkm** [comment]
here is the full screen, with bold text. no colour
![ktrussInteractionButtonBold](https://user-images.githubusercontent.com/88649439/134460569-7a805d71-b6b8-40f2-8f8a-65d035e79497.gif)

**Comment by Atlas139mkm** [comment]
With white text
![ktrussInteractionButtonBoldWhite](https://user-images.githubusercontent.com/88649439/134460746-6e459872-4e53-4872-bd0a-ff821c06fc27.gif)

**Comment by arcturus2** [comment]
Thanks @Atlas139mkm. What do you think?

I see that you have the dark theme going to white text won't work with the default theme. My vote would be to make it bold to start with unless you had other ideas?

**Comment by Atlas139mkm** [comment]
I just changed my theme back to the normal and i see the problem. 

I think bold definitely helps, its definitely better. I am drawn more to press it, however I also know to press it now. I think bolding it is a fine solution for now. The only other thing I could think of would be adding colour but if you decide you want to implement the bootstrap style buttons It could change then.

**Comment by arcturus2** [comment]
Thanks @Atlas139mkm for making the change.

I think we can look at the styling of buttons for the entire application in one go. It's one of those things that need to be done at once - otherwise it looks odd.

**Comment by GammaVel** [comment]
It has been really interesting to follow along the conversation and see how the decision was arrived at. Nice work @Atlas139mkm !
I took at look at this and I really like the bolding and the text change. In fact, I didn't even realise that the functionality hadn't changed as the button labelling seemed was more intuitive. 
![image](https://user-images.githubusercontent.com/83679004/135384173-56a0f7d9-9048-4de7-afb5-1201aa110612.png)
 
A couple of comments - I found a run time error which isn't related to the fix when there no Ktrusses. I will raise a separate bug for that. Also, there are similar buttons on the hierarchical tool which haven't changed, it might be good to make them consistent so I'll pop in an issue for that too in case you want to make them the same.
